tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py

```diff
@@ -170,95 +171,108 @@ def _forward_backward_test_impl(
 
             pp_utils.update_num_microbatches(0)
 
-            loss = fwd_bwd_func(
-                testing_utils.fwd_step_func,
-                batch,
-                model,
-                forward_only=forward_only,
-                # `tensor_shape` is the shape of micro batch.
-                tensor_shape=(
-                    self.MICRO_BATCH_SIZE,
-                    self.HIDDEN_SIZE,
-                    self.HIDDEN_SIZE,
-                ),
-                dtype=dtype,
-                async_comm=async_comm,
-                grad_scaler=grad_scaler,
-                deallocate_pipeline_output=deallocate_pipeline_outputs,
-            )
-
-            if dtype == torch.double:
-                hidden_size = self.HIDDEN_SIZE
-                microbatch_size = self.MICRO_BATCH_SIZE
-                total_layers = pipeline_model_parallel_world_size
-                if virtual_pipeline_model_parallel_size is not None:
-                    total_layers *= virtual_pipeline_model_parallel_size
-                target_loss, target_model = get_target_loss_and_model(global_batch_shape, hidden_size, total_layers)
-
-                for loss_item in loss:
-                    x = loss_item['avg']
-                    torch.testing.assert_close(x.item() / microbatch_size, target_loss.item())
+            for epoch in range(self.NUM_EPOCHS):
+                loss = fwd_bwd_func(
+                    testing_utils.fwd_step_func,
+                    batch,
+                    model,
+                    forward_only=forward_only,
+                    # `tensor_shape` is the shape of micro batch.
+                    tensor_shape=(
+                        self.MICRO_BATCH_SIZE,
+                        self.HIDDEN_SIZE,
+                        self.HIDDEN_SIZE,
+                    ),
+                    dtype=dtype,
+                    async_comm=async_comm,
+                    grad_scaler=grad_scaler,
+                    deallocate_pipeline_output=deallocate_pipeline_outputs,
+                )
+
+                if dtype == torch.double and epoch == 0:
```

Comment:
Might need to adjust this condition if one test variant has `float` specified as the dtype.